server/src/controllers/Products/IndexProductsController.js

```diff
@@ -33,9 +33,10 @@ class IndexProductsController {
                 await this.redisClientService.jsonSet(`product:${id}`, '.', JSON.stringify(product));
             }
 
-            for (const key of cartKeys) {
-                await this.redisClientService.redis.hdel(`product:${id}`, key);
-            }
+            // this piece of code is not working => id is undefined
+            // for (const key of cartKeys) {
+            //     await this.redisClientService.redis.hdel(`product:${id}`, key);
```

Comment:
this piece of code not working

Reply:
it should be `await this.redisClientService.redis.hdel(\`cart:${id}\`, key);`

Reply:
I did this bad, because I wanted to delete all carts here

Reply:
id is undefined -> so this is also error

Reply:
it has been remade